app/renderer/reducers/frameReducer.js

```diff
@@ -128,7 +128,14 @@ const frameReducer = (state, action, immutableAction) => {
           }
           state = state.setIn(['frames', index, 'lastAccessedTime'], new Date().getTime())
 
-          state = frameStateUtil.updateTabPageIndex(state, frame)
+          // Only update tabPage if user is not hovering other page
+          // Otherwise each time the active tab is updated the tabPage
+          // will be updated as well. While we want that behavior not necessarily
+          // the user will be on the same tabPage as the active tab. See bug #9779
+          const isPreviewing = state.get('frames').some(frame => frame.get('hoverState'))
```

Comment:
I think this check should be in `updateTabPageIndex` to make the functionality consistent wherever it is called, but there are also a few issues here I think. The first one is that we still want to updated the tab page index, but without clearing the `previewTabPageIndex`. Moving the check inside `updateTabPageIndex`, right? If the check is inside `updateTabPageIndex` that is easily accomplished. The second is that I think we want to check hover state for the tab pages and tabs `['ui', 'tabs', 'hoverTabPageIndex']`? The last is that we want to avoid iterating over frames and I can't see any reason to have a per-frame `hoverState` prop. Can we change it to `['ui', 'tabs, 'hoverTabIndex']`?